Tests/Marketplace/search_and_install_packs.py

```diff
@@ -314,47 +274,77 @@ def install_packs_private(client: demisto_client,
+            super().__init__()
+
+    class MalformedPackException(ApiException):
+        def __init__(self, pack_ids):
+            self.malformed_ids = pack_ids
+            super().__init__()
+
+    def call_install_packs_request(packs):
+        try:
+            logging.debug(f'Installing the following packs on server {host}:\n{[pack["id"] for pack in packs]}')
+            response_data, status_code, _ = demisto_client.generic_request_func(client,
+                                                                                path='/contentpacks/marketplace/install',
+                                                                                method='POST',
+                                                                                body={'packs': packs,
+                                                                                      'ignoreWarnings': True},
+                                                                                accept='application/json',
+                                                                                _request_timeout=request_timeout)
+
+            if status_code in range(200, 300):
+                packs_data = [{'ID': pack.get('id'), 'CurrentVersion': pack.get('currentVersion')} for pack in
+                              ast.literal_eval(response_data)]
+                logging.success(f'Packs were successfully installed on server {host}')
+                logging.debug(f'The packs that were successfully installed on server {host}:\n{packs_data}')
+
+        except ApiException as ex:
+            if 'timeout awaiting response' in ex.body:
+                raise GCPTimeOutException(ex.body)
+            if malformed_ids := find_malformed_pack_id(ex.body):
+                raise MalformedPackException(malformed_ids)
+            raise ex
+
     try:
-        response_data, status_code, _ = demisto_client.generic_request_func(client,
-                                                                            path='/contentpacks/marketplace/install',
-                                                                            method='POST',
-                                                                            body=request_data,
-                                                                            accept='application/json',
-                                                                            _request_timeout=request_timeout)
+        logging.info(f'Installing packs on server {host}')
+        logging.info(f'TESTING: adding failing pack to pack list to create failure')
+        packs_to_install.append({'id': 'PhishAI', 'version': '1.0.0'})  # TODO: remove failing pack!
+        try:
+            call_install_packs_request(packs_to_install)
+
+        except MalformedPackException as e:
+            # if this is malformed pack error, remove malformed packs and retry until success
```

Comment:
This retry is now without the malformed pack - so its likely that it will not cause an error again. 
This mechanism handles all the problematic scenarios so it will **not** result in a infinite loop:
- first iteration: the malformed pack was removed, second iteration: this pack (for some reason) still causes a failure - this will raise an Exception and break the loop (line 230).
- first iteration:  the malformed pack was removed, second iteration: a different pack causes this error - the second pack will now be removed and the request will be sent again without it. it is unlikely that this will cause iteration over all of the packs, since if there is an issue will all of the packs, the error will not include a name of a specific pack (and this means this will not raise a MalformedPackException).
- first iteration:  the malformed pack was removed, second iteration: a different error is being raised - a different error will be raised, hence there will be another 1 iteration maximum (due to the mechanism of the other errors).